pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -184,18 +184,26 @@ private TableDataManager createTableDataManager(String tableNameWithType, TableC
     return tableDataManager;
   }
 
+  @Override
+  public void deleteTable(String tableNameWithType) {
+    _tableDataManagerMap.compute(tableNameWithType, (k, v) -> {
+      if (v != null) {
+        v.shutDown();
```

Comment:
What happens to the physical segment files on the server after we call `shutdown()`?

Reply:
The files on disk will be removed when the segment goes from `OFFLINE` to `DROPPED`, the clean up is based on table and segment name. We won’t have any issue if the tableDataManager is shutdown&removed before this state transition. But we will have the issue to go from `ONLINE` to `OFFLINE`, this is because segmentDataManger is destroyed via the tableDataManager for the state transition. As you mentioned in other comments, we need to make sure all segments are `OFFLINE/DROPPED` before removing the tableDataManager. Else we may have some resources unclosed.